tests/src/Unit/SocialApiUserTest.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Drupal\Tests\social_api\Unit;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\social_api\SocialApiDataHandler;
+use Drupal\social_api\User\UserAuthenticator;
+use Drupal\social_api\User\UserManager;
+use Drupal\social_api\User\UserManagerInterface;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests social_api User.
+ *
+ * @group social_api
+ */
+class SocialApiUserTest extends UnitTestCase {
+
+  /**
+   * The tested Social Api UserManager.
+   *
+   * @var \Drupal\social_api\User\UserManager
+   */
+  protected $userManager;
+
+  /**
+   * The tested Social Api UserAuthenticator.
+   *
+   * @var \Drupal\social_api\User\UserAuthenticator
+   */
+  protected $userAuthenticator;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    $current_user = $this->createMock(AccountProxyInterface::class);
+    $data_handler = $this->createMock(SocialApiDataHandler::class);
+    $entity_type_manager = $this->createMock(EntityTypeManagerInterface::class);
+    $logger_factory = $this->createMock(LoggerChannelFactoryInterface::class);
+    $messenger = $this->createMock(MessengerInterface::class);
+    $user_manager = $this->createMock(UserManagerInterface::class);
+
+    $entity_type = 'users';
+
+    $this->userManager = $this->getMockBuilder(UserManager::class)
+      ->setConstructorArgs([$entity_type,
+        $entity_type_manager,
+        $messenger,
+        $logger_factory,
+      ])
+      ->setMethods(NULL)
+      ->getMock();
+
+    $this->userAuthenticator = $this->getMockBuilder(UserAuthenticator::class)
+      ->setConstructorArgs([$current_user,
+        $messenger,
+        $logger_factory,
+        $user_manager,
+        $data_handler,
+      ])
+      ->setMethods(NULL)
+      ->getMock();
+
+  }
+
+  /**
+   * Tests for class UserManager.
+   */
+  public function testUserManager() {
+    $this->assertTrue(
+      method_exists($this->userManager, 'getPluginId'),
+      'UserManager class does not implements getPluginId function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userManager, 'setPluginId'),
+      'UserManager class does not implements setPluginId function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userManager, 'getDrupalUserId'),
+      'UserManager class does not implements getDrupalUserId function/method'
+    );
+  }
```

Comment:
Unless the class is inheriting from an abstract class, we don't need to check this. I don't know if that's the case off the top of my head